crates/uv/tests/lock.rs

```diff
@@ -7963,7 +7963,7 @@ fn unconditional_overlapping_marker_disjoint_version_constraints() -> Result<()>
 
     ----- stderr -----
       × No solution found when resolving dependencies:
-      ╰─▶ Because your project depends on datasets<2.19 and datasets>=2.19, we can conclude that your project's requirements are unsatisfiable.
+      ╰─▶ Because only datasets<2.19 is available and your project depends on datasets>=2.19, we can conclude that your project's requirements are unsatisfiable.
     "###);
```

Comment:
Yeah — this seems wrong in this context.

Reply:
Looking at this..

Moving so we can see the display before mutation...

```diff
diff --git a/crates/uv-resolver/src/error.rs b/crates/uv-resolver/src/error.rs
index 4a38e3e30..dc9eeecf8 100644
--- a/crates/uv-resolver/src/error.rs
+++ b/crates/uv-resolver/src/error.rs
@@ -222,7 +222,6 @@ impl std::fmt::Display for NoSolutionError {
 
         // Transform the error tree for reporting
         let mut tree = self.error.clone();
-        simplify_derivation_tree_markers(&self.python_requirement, &mut tree);
         let should_display_tree = std::env::var_os("UV_INTERNAL__SHOW_DERIVATION_TREE").is_some()
             || tracing::enabled!(tracing::Level::TRACE);
 
@@ -230,6 +229,7 @@ impl std::fmt::Display for NoSolutionError {
             display_tree(&tree, "Resolver derivation tree before reduction");
         }
 
+        simplify_derivation_tree_markers(&self.python_requirement, &mut tree);
         collapse_no_versions_of_workspace_members(&mut tree, &self.workspace_members);
```

Using 

```
UV_INTERNAL__SHOW_DERIVATION_TREE=1 cit unconditional_overlapping_marker_disjoint_version_constraints
```

We get the following trees:

```
Resolver derivation tree before reduction
  root==0a0.dev0 depends on project*
      project==0.1.0 depends on datasets{python_full_version >= '3.11'}>=2.19
      no versions of datasets{python_full_version >= '3.11'}>=2.19
    no versions of project<0.1.0 | >0.1.0
Resolver derivation tree after reduction
  project==0.1.0 depends on datasets>=2.19
  no versions of datasets>=2.19
```

It looks like your simplifications here are fine, the tree is wrong before then.

Here are the (correct) trees from main:

```
Resolver derivation tree before reduction
  root==0a0.dev0 depends on project*
      project==0.1.0 depends on datasets>=2.19
      project==0.1.0 depends on datasets<2.19
    no versions of project<0.1.0 | >0.1.0
Resolver derivation tree after reduction
  project==0.1.0 depends on datasets>=2.19
  project==0.1.0 depends on datasets<2.19
```